analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Nullable.cs

```diff
@@ -77,13 +77,12 @@ public void Nullable_HasValue_ReadsBoolConstraintFromObjectConstraint()
             Tag("SymbolAfterNull", arg);
             """;
         var validator = SETestContext.CreateCS(code, ", bool? arg").Validator;
-        validator.ValidateTag("HasValueUnknown", x => x.HasConstraint<BoolConstraint>().Should().BeFalse());
-        validator.ValidateTag("SymbolUnknown", x => x.Should().BeNull());
-        validator.ValidateTag("HasValueAfterTrue", x => x.HasConstraint<BoolConstraint>().Should().BeFalse());   // ToDo: Should be x.HasConstraint(BoolConstraint.True).Should().BeTrue()); after we build NotNull for bool literal.
-        validator.ValidateTag("SymbolAfterTrue", x => x.HasConstraint(BoolConstraint.True).Should().BeTrue());
-        validator.ValidateTag("SymbolAfterTrue", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeFalse());    // ToDo: Should be BeTrue() after we build NotNull for bool literal
-        validator.ValidateTag("HasValueAfterNull", x => x.HasConstraint(BoolConstraint.False).Should().BeTrue());
-        validator.ValidateTag("SymbolAfterNull", x => x.HasConstraint(ObjectConstraint.Null).Should().BeTrue());
+        validator.ValidateTag("HasValueUnknown", x => x.Should().HaveOnlyConstraint(ObjectConstraint.NotNull));
+        validator.ValidateTag("SymbolUnknown", x => x.Should().HaveNoConstraints());
+        validator.ValidateTag("HasValueAfterTrue", x => x.Should().HaveOnlyConstraints(ObjectConstraint.NotNull, BoolConstraint.True));
```

Comment:
There's a rebase conflict. I think I've rewritten this UT to use `int?` instead, to avoid this pitfall. So you can probably rebase and revert these changes. Or re-add this UT as a 2nd test for this.

I don't have a preference.

Reply:
I added the `bool?` test (out of curiosity if everything still works).